src/components/Trade/TradeConfirm/TradeConfirm.tsx

```diff
@@ -262,19 +263,13 @@ export const TradeConfirm = () => {
       // Poll until we have a "buy" txid
       // This means the trade is just about finished
       const txs = await poll({
-        fn: async () => {
-          try {
-            return { ...(await swapper.getTradeTxs(result.unwrap())) }
-          } catch (e) {
-            return { sellTxid: '', buyTxid: '', e }
-          }
-        },
-        validate: (txs: TradeTxs & { e: Error }) => !!txs.buyTxid || !!txs.e,
+        fn: () => swapper.getTradeTxs(result.unwrap()).catch(e => Err(e)),
+        validate: txsResult => txsResult.isOk() && !!txsResult.unwrap().buyTxid,
         interval: 10000, // 10 seconds
         maxAttempts: 300, // Lots of attempts because some trade are slow (thorchain to bitcoin)
       })
-      if (txs.e) throw txs.e
-      setBuyTxid(txs.buyTxid ?? '')
+      if (txs.isErr()) throw txs.err()
```

Comment:
@woodenfurniture `err()` this will throw a `Some<SwapErrorRight>` instead of a `SwapErrorRight`, and I don't see us unwrapping the `Option` in the catch - are we sure we don't want to use `unwrapErr()`?